src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -29,28 +30,36 @@ public List<Faculty> getAllFaculty() {
         return service.getAllFaculty();
     }
 
-
-    @GetMapping("id")
-    public Faculty findById(@RequestParam("id") long id) {
+    @GetMapping("{id}")
+    public Faculty findById(@PathVariable("id") long id) {
         return service.findById(id);
     }
 
-
-    @GetMapping("search")
+    @GetMapping(params = "color")
     public List<Faculty> searchByColor(@RequestParam("color") String color) {
         return service.searchByColor(color);
     }
 
+    @GetMapping(params = "part")
+    public List<Faculty> searchByPart(@RequestParam("part") String part) {
+        return service.searchByParty(part);
+    }
+
+    @GetMapping("/getStudents/{facultyId}")
```

Comment:
сначала обращаемся к факультету по id и затем берём его свойство - студентов
```suggestion
    @GetMapping("/{id}/students")
```

Reply:
Сегодня же такая мысль посетила, иерархически более верно, но уже отправил на проверку.
Вчера голова забита была борьбой с ошибками в эндпойнте получения факультета студента при fetch=LAZY в модели. Причем факультет приходил в эндпойнт (проверил выводом sout в консоль), а в API - Error 500 и длинная ошибкой в консоли. Принципиально не хотел использовать DTO, решение должно быть простым. В итоге нашел решение (@JsonIgnoreProperties), но его механика остается весьма туманной

Reply:
Ага, ну рекурсивный вывод сущностей я успешно купировал @JsonIgnore, но получил ошибку вызова факультета.  Поборол этой аннотацией, совместно с @JsonBackReference.  
Если DTO правильное решение, то, может, надо было внести его в ДЗ? 
Транзакции мы еще не изучали, насколько помню, поэтому суть остается туманной, возможно позже прояснится.
И, как я понял, DTO теряет часть сущности, т.е. запрос получает не всю сущность. Это нормальное решение? Или здесь отбрасывается только излишняя, для данного конкретного запроса, часть сущности?

Reply:
Дмитрий, спасибо за ответ. 
Решил разобраться с DTO, попробовал реализовать слой с ручным копированием полей. В сети видел разные подходы к DTO, там кто во что горазд. Решил: обработку DTO расположить в классах контроллеров, использовать ручное создание (для начала). Автомапперы пока не использовал.
Все работает, как предополагалось.
Насколько верно реализованы мной DTO?

Reply:
> Решил: обработку DTO расположить в классах контроллеров

В контроллерах не должно быть никакой логики. Маппинг - это тоже логика. Лучше вынести в отдельные классы-компоненты - мапперы

Reply:
Все реализовал. Теперь контроллер и сервис обмениваются данными Student и Faculty через посредников в виде доп классов (кроме удаления - там void, передавать нечего) . Управляет наполнением объектов в обе стороны маппер, каждый для своего класса. Теперь можно убрать вызовы маппера из контроллера и работать с сервисом напрямую, если возникнет такая необходимость. Понимаю, что это в большей степени "ручной привод", но в учебных целях пока сойдет